harvard-lts/hdc3a-dlq-handler-service
# Admin email notifications for DLQ message handling events

**Admin email notifications for DLQ message handling events**
* * *

**GitHub Issue**: ([link](https://app.zenhub.com/workspaces/hdc-62163f60fb08c60011fa4795/issues/harvard-lts/hdc/217))

# What does this Pull Request do?
Adds admin email notifications for DLQ message handling events. In particular, the following events are notified:

- Missing required fields in the body dict of a received message
- Failed to resubmit a message to its original queue
- Message resubmitted
- The maximum number of retry attempts for a message has been reached

# How should this be tested?

A description of what steps someone could take to:

* Update test .env file with the new emailing environment variables
* Rebuild test environment: docker-compose -f docker-compose-test.yml build --no-cache
* Run the local test environment: `docker-compose -f docker-compose-test.yml up`
* Execute tests: `docker exec -it test_runner pytest test`

For a functional test:

* Update .env file with the new emailing environment variables
* Follow the same steps done in PR #2 functional test, and check that email messages are received related to the corresponding retrying events.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests: Yes
- integration tests: Yes

# Interested parties
@jessjass @ives1227 @adaybujeda 

Keyword: resolved
